### ISSUE TITLE
Reproducible builds and vulnerabilities

### ISSUE THREAD

**Issue by samontab** [buttercup/buttercup-desktop#1193]
Thanks for making this great software.

Before testing this further, I wanted to build a matching binary as the one posted in the release.

Here's the sha256sum from the release v2.18.2:

```
wget https://github.com/buttercup/buttercup-desktop/releases/download/v2.18.2/Buttercup-linux-x86_64.AppImage
sha256sum Buttercup-linux-x86_64.AppImage
```
The output is:
`d562e14fde05359edb7b9f2ea2fff4bedbc1ec44ff58d63fb4f488b9a5b9cdf7  Buttercup-linux-x86_64.AppImage`

Here are the steps I used to build the AppImage locally:
```
git clone --branch v2.18.2 https://github.com/buttercup/buttercup-desktop.git
cd buttercup-desktop
npm install
npm run build
npm run package:linux
sha256sum dist/Buttercup-linux-x86_64.AppImage
```

In the readme of the project it says:

> To begin developing features or bug-fixes for Buttercup Desktop, make sure that you first have Node v14 installed with a current version of NPM.

In the package.json file it says:

```
  "engines": {
    "node": ">=16",
    "npm": ">=7"
```
Also, the current node LTS is v18, so I tried with all three of them. It builds fine in all cases, but it gives me a different value than the published one.

```
node v14.21.3
npm v6.14.18
1bbea22eef4c9e7eccbb424a79e17c8599ade579018da94c87ecb821f0bcfbaf  dist/Buttercup-linux-x86_64.AppImage
```

```
node v16.19.1
npm v8.19.3
e5aeca3dd30502d190bf2b572a72d96d023603043ab8d171811d39ff37df2412  dist/Buttercup-linux-x86_64.AppImage
```

```
node v18.15.0
npm v9.5.0
33ac0366610fdd239f6a8fed7a6affc26b88e380049fd0a8a8adb5590f084dec  dist/Buttercup-linux-x86_64.AppImage
```

So, how can I reproduce the published binaries?

Also, it's worth noting that in the current source code, npm listed some vulnerabilities:

> node v14: found 24 vulnerabilities (3 low, 1 moderate, 20 high)
> node v16: 19 vulnerabilities (1 low, 4 moderate, 14 high)
> node v18: 19 vulnerabilities (1 low, 4 moderate, 14 high)

npm audit fix seems to remove some of them, but there were others that might need some attention.

**Comment by perry-mitchell** [comment]
Thanks for pointing these out. I'd gladly accept PRs to fix the documentation and to address the vulnerabilities.. though some of them might require minor/major dependency upgrades with further testing needed.

**Comment by Oxalin** [comment]
The following takes care of the README.md file
https://github.com/Oxalin/buttercup-desktop/commit/f022f290d87b2506566bb559b18af5b7c7a836df

**Comment by Oxalin** [comment]
If I may: I would suggest to stick with a Node LTS version for the test and development purposes and bump to the next LTS once in a while. This would partially answer build reproductibility raised by @samontab.

However, it should be possible to rebuild a release mostly from package-lock.json (as long as the release tag and the binaries go with one another).

**Comment by perry-mitchell** [comment]
I agree with the above.. Given the same Node JS version, and the committed package.json and package-lock.json files, builds should already be reproducible. The documentation should be better of course. I personally always use LTS versions but that should be documented as the only supported platform.
